src/item/enchantment/ItemEnchantmentTagRegistry.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\item\enchantment;
+
+use pocketmine\item\enchantment\ItemEnchantmentTags as Tags;
+use pocketmine\utils\SingletonTrait;
+use function array_diff;
+use function array_intersect;
+use function array_merge;
+use function array_search;
+use function array_unique;
+use function count;
+
+final class ItemEnchantmentTagRegistry{
+	use SingletonTrait;
+
+	/** @phpstan-var array<string, string[]> */
```

Comment:
```suggestion
	/** @phpstan-var array<string, list<string>> */
```

Reply:
This should also be accompanied by a conventional `@var string[][]` for tools that don't understand non-PSR PHPDoc